backend/modules/simulation/agents/grading_agent.py

```
@@ -4,6 +4,8 @@
 """
 
 import json
+import re
+from typing import Dict, List, Any, Optional
 from typing import Dict, List, Any, Optional, Tuple
```

Ruff (0.15.4)

[error] Redefinition of unused `Dict` from line 8: `Dict` redefined here

Remove definition: `Dict`

(F811)